src/com/cft/contactmerge/NameMatchLogic.java

```diff
@@ -0,0 +1,129 @@
+/* Created by James Page on May 8th, 2018
+ * Contains the low level business logic for matching naming properties
+ */
+
+package com.cft.contactmerge;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public abstract class NameMatchLogic {
+    /***********************************************************************
+     ****************************** Variables ******************************
+     ***********************************************************************/
+    private final static int hyphenCharLimitYesVsMaybe = 3;
+    private final static String[] namePunctuations = {".", ",", "'", "-"};
+    protected final static String[][] alternateFirstNames = // List of names to match nicknames of the same person
```

Comment:
Are we already using this list? If not, would remove it for now and we can add a backlog item.  If it is used, we need a  TODO comment indicating that the list is incomplete and we have to find a source for the information as creating one ourselves would be tough.

Reply:
Will remove it for now. Can always re-create later if we decide we need it.